packages/tangle/utils.go

```diff
@@ -123,24 +125,17 @@ func (u *Utils) TransactionApprovedByMessage(transactionID ledgerstate.Transacti
 		}
 
 		bookedParents := make(MessageIDs, 0)
-		u.tangle.Storage.Message(messageID).Consume(func(message *Message) {
-			for _, parentID := range message.StrongParents() {
-				var parentBooked bool
-				u.tangle.Storage.MessageMetadata(parentID).Consume(func(parentMetadata *MessageMetadata) {
-					parentBooked = parentMetadata.IsBooked()
-				})
-				if !parentBooked {
-					continue
-				}
-
-				// First check all of the parents to avoid unnecessary checks and possible walking.
-				if attachmentMessageID == parentID {
-					approved = true
-					return
-				}
 
-				bookedParents = append(bookedParents, parentID)
+		u.tangle.Storage.Message(messageID).Consume(func(message *Message) {
+			approved = u.checkBookedParents(message, &attachmentMessageID, func(message *Message) MessageIDs {
```

Comment:
since message and attachment message are immutable I am not sure this is an issue...
will only fix if I am being asked to :angel:

Reply:
Even if attachmentMessageID's value is immutable, it could lead to some unexpected behavior since the address of the loop variable doesn't change but its assigned value does. example: https://husni.dev/beware-of-implicit-memory-aliasing-in-go-foor-loop/

Doesn't look to be an issue how it's used here, but just fyi.

Reply:
Thanks

Now that I understand the issue better I will fix this.
What is not an issue now may become an issue after someone else in the future changes the code slightly..
I should write secure code to begin with :-)